src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Cohort/ReviewApprenticeRequestToReviewApprenticeViewModelMapper.cs

```diff
@@ -18,68 +18,101 @@ public class ReviewApprenticeRequestToReviewApprenticeViewModelMapper : IMapper<
+
+            //Get commitments from DB
             if (!string.IsNullOrWhiteSpace(result.CohortRef))
             {
-                var commitmentId = _encodingService.Decode(result.CohortRef, EncodingType.CohortReference);
-                result.MessageFromEmployer = (await _commitmentsApiClient.GetCohort(commitmentId)).LatestMessageCreatedByEmployer;
+                var cohortId = _encodingService.Decode(source.CohortRef, EncodingType.CohortReference);
+                result.MessageFromEmployer = (await _commitmentsApiClient.GetCohort(cohortId)).LatestMessageCreatedByEmployer;
+                var response = await _commitmentsApiClient.GetDraftApprenticeships(cohortId);
+                if (response != null)
+                {
+                    _logger.LogInformation("Total number of records from db: " + response.DraftApprenticeships.Count);
+                    await ExistingCohortDetails(result, response);
+                }
             }
 
+            var publicAccountLegalEntityIdNew = _encodingService.Decode(csvRecordsGroupedByCohort.FirstOrDefault().AgreementId, EncodingType.PublicAccountLegalEntityId);
+            result.EmployerName = (await _commitmentsApiClient.GetAccountLegalEntity(publicAccountLegalEntityIdNew)).AccountName;
+            
             return result;
+        }       
+
+        private async Task FileUploadCohortDetails(FileUploadReviewApprenticeViewModel result, IEnumerable<CsvRecord> csvRecordsGroupedByCohort)
+        {
+            foreach (var csvRecord in csvRecordsGroupedByCohort)
+            {
+                var courseDetails = await _commitmentsApiClient.GetTrainingProgramme(csvRecord.StdCode);
+                var dateOfBirth = GetValidDate(csvRecord.DateOfBirth, "yyyy-MM-dd");
+                var apprenticeStartDate = GetValidDate(csvRecord.StartDate, "yyyy-MM-dd");
+                var apprenticeEndDate = GetValidDate(csvRecord.EndDate, "yyyy-MM");
+
+                var fileUploadApprenticeDetail = new ReviewApprenticeDetailsForFileUploadCohort
+                {
+                    Name = $"{csvRecord.GivenNames} {csvRecord.FamilyName}",
+                    TrainingCourse = courseDetails.TrainingProgramme.Name,
+                    ULN = csvRecord.ULN,
+                    DateOfBirth = dateOfBirth.Value.ToString("d MMM yyyy"),
+                    Email = csvRecord.EmailAddress,
+                    TrainingDates = $"{apprenticeStartDate.Value:MMM yyyy} to {apprenticeEndDate.Value:MMM yyyy} ",
+                    Price = int.Parse(csvRecord.TotalPrice),
+                    FundingBandCap = GetFundingBandCap(courseDetails.TrainingProgramme, apprenticeStartDate.Value.Date)
+                };
+
+                result.FileUploadCohortDetails.Add(fileUploadApprenticeDetail);
+            }
+        }
+
+        private async Task ExistingCohortDetails(FileUploadReviewApprenticeViewModel result, CommitmentsV2.Api.Types.Responses.GetDraftApprenticeshipsResponse response)
```

Comment:
rather than passing in result by reference and populating a property, it might be simpler to return `Task<List<ReviewApprenticeDetailsForExistingCohort>>` and set that on the response object at the end of the mapping